HW3/Eddy/question2-3.js

```diff
@@ -0,0 +1,17 @@
+const mergeAndSortArrays = (arr1, arr2) => {
+    const mergedMap = arr1.reduce((acc, item) => {
+      acc[item.uuid] = { ...item, role: null };
+      return acc;
+    }, {});
+  
+    arr2.forEach(item => {
+      mergedMap[item.uuid] = { ...mergedMap[item.uuid], ...item, name: mergedMap[item.uuid]?.name || null };
```

Comment:
```suggestion
      mergedMap[item.uuid] = { ...item, name: mergedMap[item.uuid]?.name || null};
````
`...mergedMap[item.uuid]`is not necessary here. `...item` has uuid property